src/data/list/basic.lean

```diff
@@ -1718,6 +1718,23 @@ lemma nth_le_drop' (L : list α) {i j : ℕ} (h : j < (L.drop i).length) :
   nth_le (L.drop i) j h = nth_le L (i + j) (nat.add_lt_of_lt_sub_left ((length_drop i L) ▸ h)) :=
 by rw nth_le_drop
 
+lemma nth_drop' (L : list α) (i j : ℕ) :
+  nth (L.drop i) j = nth L (i + j) :=
+begin
+  ext,
+  simp only [nth_eq_some, nth_le_drop', option.mem_def],
+  split,
+  { rintro ⟨h, rfl⟩,
+    simp only [length_drop] at h,
+    simpa using nat.add_lt_of_lt_sub_left h },
+  { rintro ⟨h, rfl⟩,
+    simpa using nat.lt_sub_left_of_add_lt h }
+end
+
+lemma nth_drop (L : list α) (i j : ℕ) :
+  nth L (i + j) = nth (L.drop i) j :=
+by rw nth_drop'
```

Comment:
Why do you need both version? The second is just `(nth_drop _ _ _).symm`

Reply:
Both versions are to replicate the `list.nth_le_drop(')` lemmas right above.

Reply:
The only reason the lemmas above are done that way is because the lemma statement contains a proof obligation, which means it can't just be reversed, as the derived proof on the RHS won't match a proof derived a different way (apparently).

That's not relevant for your lemma.

Reply:
Good point! The direction the lemma name implies is the one I've retained.